File: 1007-minimum-domino-rotations-for-equal-row/1007-minimum-domino-rotations-for-equal-row.java
class Solution {
    public int minDominoRotations(int[] tops, int[] bottoms) {
        int a = tops[0], b = bottoms[0];
        int n = tops.length;
        
        for(int i=1; i < n; i++) {
            if(a != tops[i] && a != bottoms[i]) {
                a = -1;
            }
            if(b != tops[i] && b != bottoms[i]) {
                b = -1;
            }
        }
        if(a == -1 && b == -1) 
            return -1;
        
        Integer ans1 = Integer.MAX_VALUE;
        Integer ans2 = Integer.MAX_VALUE;
        if(a != -1) {
            int topC=0, bottomC=0;
            for(int i=0; i<n; i++) {
                if(tops[i] == a) 
                    topC++;
                if(bottoms[i] == a)
                    bottomC++;
            }
            ans1 = n - Math.max(topC, bottomC);
        }
        if(b != -1) {
            int topC=0, bottomC=0;
            for(int i=0; i<n; i++) {
                if(tops[i] == b) 
                    topC++;
                if(bottoms[i] == b)
                    bottomC++;
            }
            ans2 = n - Math.max(topC, bottomC);
        }
        return Math.min(ans1, ans2);
    }
}

File: 1007-minimum-domino-rotations-for-equal-row/1007-minimum-domino-rotations-for-equal-row.py
class Solution:
    def minDominoRotations(self, tops: List[int], bottoms: List[int]) -> int:
        n = len(tops)
        # Find the element common among all the dices
        a, b = tops[0], bottoms[0]
        for i in range(1, n):
            if a not in (tops[i], bottoms[i]):
                a = -1
            if b not in (tops[i], bottoms[i]):
                b = -1
        if a == -1 and b == -1:
            return -1
        
        ans1, ans2 = float('inf'), float('inf')
        if a != -1:
            # We count the top occurences of a and bottom occurences of a
            topCount = 0
            for i in tops:
                if i == a:
                    topCount += 1
            bottomCount = 0
            for i in bottoms:
                if i == a:
                    bottomCount += 1
            ans1 = n - max(topCount, bottomCount)                        
        if b != -1:
            topCount = 0
            for i in tops:
                if i == b:
                    topCount += 1
            bottomCount = 0
            for i in bottoms:
                if i == b:
                    bottomCount += 1
            ans2 = n - max(topCount, bottomCount)
        return min(ans1, ans2)